includes/sanitizers/class-amp-blacklist-sanitizer.php

```diff
@@ -93,24 +110,68 @@ private function sanitize_a_attribute( $node, $attribute ) {
 			// rev removed from HTML5 spec, which was used by Jetpack Markdown.
 			$node->removeAttribute( $attribute_name );
 		} elseif ( 'target' === $attribute_name ) {
-		       if ( '_blank' === $attribute->value || '_new' === $attribute->value ) {
-			       // _new is not allowed; swap with _blank
-			       $node->setAttribute( $attribute_name, '_blank' );
-		       } else {
-			       // only _blank is allowed
-			       $node->removeAttribute( $attribute_name );
-		       }
+			// _blank is the only allowed value and it must be lowercase.
+			// replace _new with _blank and others should simply be removed.
+			$old_value = strtolower( $attribute->value );
+			if ( '_blank' === $old_value || '_new' === $old_value ) {
+				// _new is not allowed; swap with _blank
+				$node->setAttribute( $attribute_name, '_blank' );
+			} else {
+				// only _blank is allowed
+				$node->removeAttribute( $attribute_name );
+			}
+		} elseif ( 'href' === $attribute_name ) {
+			// If the href starts with a '/', append the home_url to it for validation purposes.
+			$href = $attribute->value;
+			if ( 0 === stripos( $attribute->value, '/' ) ) {
+				$href = untrailingslashit( get_home_url() ) . $href;
+			}
+
+			$valid_protocols = array( 'http', 'https', 'mailto', 'sms', 'tel', 'viber', 'whatsapp' );
+			$protocol = strtok( $href, ':' );
+			if ( false === filter_var( $href, FILTER_VALIDATE_URL )
+				|| ! in_array( $protocol, $valid_protocols ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	private function replace_node_with_children( $node ) {
+		// If the node has children and also has a parent node,
+		// clone and re-add all the children just before current node.
+		if ( $node->hasChildNodes() && $node->parentNode ) {
+			foreach ( $node->childNodes as $child_node ) {
+				$new_child = $child_node->cloneNode( true );
+				$node->parentNode->insertBefore( $new_child, $node );
+			}
+		}
+
+		// Remove the node from the parent, if defined.
+		if ( $node->parentNode ) {
```

Comment:
Yeah, this is an important sanity check in my opinion. Some weird, invalid HTML could in theory cause a fatal error here. There's no harm in having this check and I think we should leave it.